src/gc.c

```diff
@@ -133,8 +133,7 @@ void gc_sweep(struct gc_arena *a, size_t start) {
   for (cur = a->base + (start * a->cell_size);
        cur < (a->base + (a->size * a->cell_size));
        cur += a->cell_size) {
-    uintptr_t it = (* (uintptr_t *) cur);
-    if (it & 1) {
+    if (MARKED(cur)) {
```

Comment:
crash when GC is enabled is caused by this change.

`MARKED(x)` checks whether the pointer `x` is marked. 

`cur` is a pointer to a cell; the cell is marked if the first pointer sized word is marked:

```c
if (MARKED((* (uintptr_t *) cur))) {
```